File: app/src/main/java/com/tomclaw/drawa/stock/StockAdapter.java
package com.tomclaw.drawa.stock;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.ViewGroup;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by solkin on 19/12/2017.
 */
public class StockAdapter extends RecyclerView.Adapter<StockItemHolder> {

    private Context context;
    private final List<StockItem> items;
    private StockItemClickListener listener;

    public StockAdapter(Context context) {
        this.context = context;
        this.items = new ArrayList<>();
        setHasStableIds(true);
    }

    public void setItems(List<StockItem> items) {
        this.items.clear();
        this.items.addAll(items);
    }

    public void setClickListener(StockItemClickListener listener) {
        this.listener = listener;
    }

    @Override
    public StockItemHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        StockItemView stockView = null; // StockItemView_.build(context);
        return new StockItemHolder(stockView);

    }

    @Override
    public void onBindViewHolder(StockItemHolder holder, int position) {
        StockItem item = items.get(position);
        holder.bind(item, listener);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public int getItemCount() {
        return items.size();
    }
}

File: app/src/main/java/com/tomclaw/drawa/stock/StockAdapter.kt
package com.tomclaw.drawa.stock

import android.content.Context
import androidx.recyclerview.widget.RecyclerView
import android.view.LayoutInflater
import android.view.ViewGroup
import com.jakewharton.rxrelay2.PublishRelay
import com.tomclaw.drawa.R
import com.tomclaw.drawa.util.DataProvider

class StockAdapter(
        private val context: Context,
        private val dataProvider: DataProvider<StockItem>
) : androidx.recyclerview.widget.RecyclerView.Adapter<StockItemHolder>() {

    var itemsRelay: PublishRelay<StockItem>? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): StockItemHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.stock_item_view, parent, false)
        return StockItemHolder(view, itemsRelay)
    }

    override fun onBindViewHolder(holder: StockItemHolder, position: Int) {
        val item = dataProvider.getItem(position)
        holder.bind(item)
    }

    override fun getItemId(position: Int): Long = dataProvider.getItem(position).id.toLong()

    override fun getItemCount(): Int = dataProvider.size()

}
